packages/components/src/border-control/border-control/component.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * Internal dependencies
+ */
+import BorderDropdown from '../border-dropdown';
+import UnitControl from '../../unit-control';
+import RangeControl from '../../range-control';
+import { HStack } from '../../h-stack';
+import { StyledLabel } from '../../base-control/styles/base-control-styles';
+import { View } from '../../view';
+import { VisuallyHidden } from '../../visually-hidden';
+import { contextConnect, WordPressComponentProps } from '../../ui/context';
+import { useBorderControl } from './hook';
+
+import type { BorderControlProps, BorderLabelProps } from '../types';
+
+const BorderLabel = ( props: BorderLabelProps ) => {
+	const { label, hideLabelFromVision } = props;
+
+	if ( ! label ) {
+		return null;
+	}
+
+	return hideLabelFromVision ? (
+		<VisuallyHidden as="label">{ label }</VisuallyHidden>
+	) : (
+		<StyledLabel>{ label }</StyledLabel>
+	);
+};
+
+const BorderControl = (
+	props: WordPressComponentProps< BorderControlProps, 'div' >,
+	forwardedRef: React.Ref< any >
+) => {
+	const {
+		colors,
+		disableCustomColors,
+		enableAlpha,
+		hideLabelFromVision,
+		innerWrapperClassName,
+		label,
+		onBorderChange,
+		onSliderChange,
+		onWidthChange,
+		placeholder,
+		previousStyleSelection,
+		showStyle = true,
+		sliderClassName,
+		value: border,
+		width,
+		widthControlClassName,
+		widthUnit,
+		widthValue,
+		withSlider,
+		__experimentalHasMultipleOrigins,
+		__experimentalIsRenderedInSidebar,
+		...otherProps
+	} = useBorderControl( props );
+
+	return (
+		<View { ...otherProps } ref={ forwardedRef }>
+			<BorderLabel
+				label={ label }
+				hideLabelFromVision={ hideLabelFromVision }
+			/>
+			<HStack spacing={ 3 }>
+				<HStack className={ innerWrapperClassName } style={ { width } }>
```

Comment:
We should avoid using `style`, and instead use `width` in the `useBorderControl`, and add it to the resulting `innerWrapperClassName` (no need to make this change now, but we'd definitely need to do it before merging — I can also provide an example snippet to help! )

Reply:
I've updated this PR to avoid the inline style for width. If the `isCompact` or `width` props are present, I pass the appropriate value through to a new function returning the required CSS to roll into the inner wrapper's dynamic class.

https://github.com/WordPress/gutenberg/pull/37769/commits/1b40bb832bfa14b125232cf2cbdd99aaaf5952e8

Let me know if there is a better approach or if you had other ideas.

Reply:
Given the above-mentioned approach ticks the box for removing the inline width style. I'll mark this as resolved and it can be refined in follow-ups if required.